### DISCUSSION TITLE
Plugins in Nuxt3.

### DISCUSSION THREAD

**Discussion by HiddenUndefined** [nuxt/nuxt#18380] [Questions] [answered]
<migrated from nuxt/framework>

**Comment by HiddenUndefined** [comment, marked as answer]
I find the problem, it was that I made a mistake in the name of the plugins directory...

After renaming directory on plugins my plugin just work :)

It's correct code:
```ts
import { defineNuxtPlugin } from '#app'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.provide('hello', (msg) => `Hello ${msg}!`)
})
```

**Reply by danielroe** [reply]
For the benefit of others, would you share the issue/solution?

**Reply by dammy001** [reply]
Yes, please.

**Reply by HiddenUndefined** [reply]
The problem was that I made a mistake in the name of the `plugins` directory...

After renaming directory on `plugins` my plugin just work :)

It's correct code:
```ts
import { defineNuxtPlugin } from '#app'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.provide('hello', (msg) => `Hello ${msg}!`)
})
```
